pkg/resource/resource.go

```diff
@@ -632,8 +462,194 @@ func sendNodesMetrics(obj *unstructured.Unstructured, crName string) {
+			return fmt.Errorf("couldn't update ResourceVersion: %w", err)
+		}
+
+	}
+	if kind == "Service" {
+		clusterIP, fnd, err := unstructured.NestedString(found.Object, "spec", "clusterIP")
+		if err != nil || !fnd {
+			return fmt.Errorf("error or clusterIP not found: %w", err)
+		}
+
+		if err = unstructured.SetNestedField(req.Object, clusterIP, "spec", "clusterIP"); err != nil {
+			return fmt.Errorf("couldn't update clusterIP: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func SetNodeSelectorTerms(obj *unstructured.Unstructured, terms map[string]string) error {
+
+	if strings.Compare(obj.GetKind(), "DaemonSet") == 0 ||
+		strings.Compare(obj.GetKind(), "Deployment") == 0 ||
+		strings.Compare(obj.GetKind(), "Statefulset") == 0 { // TODO(qbarrand) should this be StatefulSet?
+		if err := nodeSelectorTerms(terms, obj, "spec", "template", "spec", "nodeSelector"); err != nil {
+			return fmt.Errorf("cannot setup %s nodeSelector: %w", obj.GetKind(), err)
+		}
+	}
+	if strings.Compare(obj.GetKind(), "Pod") == 0 {
+		if err := nodeSelectorTerms(terms, obj, "spec", "nodeSelector"); err != nil {
+			return fmt.Errorf("cannot setup Pod nodeSelector: %w", err)
+		}
+	}
+	if strings.Compare(obj.GetKind(), "BuildConfig") == 0 {
+		if err := nodeSelectorTerms(terms, obj, "spec", "nodeSelector"); err != nil {
+			return fmt.Errorf("cannot setup BuildConfig nodeSelector: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func nodeSelectorTerms(terms map[string]string, obj *unstructured.Unstructured, fields ...string) error {
+
+	nodeSelector, found, err := unstructured.NestedMap(obj.Object, fields...)
+	if err != nil {
+		return err
+	}
+
+	if !found {
+		nodeSelector = make(map[string]interface{})
+	}
+
+	for k, v := range terms {
+		nodeSelector[k] = v
+	}
+
+	if err = unstructured.SetNestedMap(obj.Object, nodeSelector, fields...); err != nil {
+		return fmt.Errorf("cannot update nodeSelector for %s : %w", obj.GetName(), err)
+	}
+
+	return nil
+}
+
+func IsOneTimer(obj *unstructured.Unstructured) (bool, error) {
+
+	// We are not recreating Pods that have restartPolicy: Never
+	if obj.GetKind() == "Pod" {
+		restartPolicy, found, err := unstructured.NestedString(obj.Object, "spec", "restartPolicy")
+		if err != nil || !found {
+			return false, fmt.Errorf("error or restartPolicy not found: %w", err)
+		}
+
+		if restartPolicy == "Never" {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func SetMetaData(obj *unstructured.Unstructured, nm string, ns string) {
+
+	annotations := obj.GetAnnotations()
+
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	annotations["meta.helm.sh/release-name"] = nm
+	annotations["meta.helm.sh/release-namespace"] = ns
+
+	obj.SetAnnotations(annotations)
+
+	labels := obj.GetLabels()
+
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	labels["app.kubernetes.io/managed-by"] = "Helm"
+
+	obj.SetLabels(labels)
+}
+
+func BeforeCRUD(obj *unstructured.Unstructured, sr interface{}) error {
```

Comment:
lets make a private function

Reply:
We agreed to address this with https://issues.redhat.com/browse/MGMT-8567